### ISSUE TITLE
Variable Explorer Accessibility Issues

### ISSUE THREAD

**Issue by AlexanderSher** [Microsoft/RTVS#3393]
**HC theme support:**
- Value edit textboxes aren't in compliance with HC themes
- Bug 402111: Accessibility : MAS42A : High Contrast: Environment selection from combo box is not in compliance with HC themes
- Bug 402112: Accessibility : MAS42A : High Contrast: Select Environment combo box when focused using keyboard doesn't get a highlight

**Keyboard navigation**
- Bug 402114: Accessibility : MAS05A : Keyboard : Focus is not visible on the cells present in the 'Variable Explorer' window
- Bug 402115: Accessibility : MAS36 : Keyboard : Table headers is not accessible in Variable Explorer window
- Bug 402116: Accessibility : MAS36 : Keyboard : Options in the 'Selected Environment' combo box in Variable Explorer window is not invoked using Space/Enter key
- Bug 402117: Accessibility : MAS05 : Keyboard : Focus order is incorrect in Variable Explorer window
- Bug 402118: Accessibility : MAS05 : Keyboard : Focus moves to the top of the window on deleting a variable in 'Variable Explorer' window
- Bug 402120: Accessibility : MAS36 : Keyboard : Table headers are not accessible in R Data window

**Comment by AlexanderSher** [comment]
Generally speaking, there is one big issue with Variable Explorer that causes all those bugs: it uses `DataGrid`. While `DataGrid` is a powerful control, it is really hard to get customized for anything expect showing table-style data. In our case, we have two columns:
- first column that isn't editable, but has a tree view logic and custom arrow key handling
- second column is partially editable and also contains additional controls on top of its base with custom tab order

**Comment by AlexanderSher** [comment]
Additional question about grid headers. It is uncommon that a grid header is part of a tab order - no existing window in VS allows to user to navigate to the header with tabs or arrows.

**Comment by MikhailArkhipov** [comment]
I think it is for sorting

**Comment by AlexanderSher** [comment]
Are there any common shortcuts for sorting?

**Comment by huguesv** [comment]
File Explorer lets me tab to the header, then arrows to move between columns and space to sort by the selected column.
![image](https://cloud.githubusercontent.com/assets/1696845/24727843/6196d898-1a0c-11e7-85ae-9f67cde6a293.png)

**Comment by int19h** [comment]
I wonder if we can get most or all of this for free if we use ListView rather than DataGrid as a basis for this UI (we'd still have to customize it into a tree, but still).

**Comment by MikhailArkhipov** [comment]
We can but prob not for 2017 at the moment. We can file for exception and then work to replace VE and grid by something else.

**Comment by AlexanderSher** [comment]
Hm... they have a bit different behavior (though probably the right one):|
- tab doesn't switch between columns, rows or cells - it only switches between header and list itself
- enter & space change sorting and switch selection back to the last selected row in the grid

To implemen this behavior, we have to get rid of the `DataGrid` and use something more customizable - in range from `ListView` to `ItemsControl`

**Comment by AlexanderSher** [comment]
> We can file for exception and then work to replace VE and grid by something else.

That's probably the best option for now. Changing anything in VE is quite risky cause UI is tightly bound to the underlying business logic - we can't simply replace `DataGrid` with `ListView`.

**Comment by MikhailArkhipov** [comment]
HC went #3408, the rest is for exception

**Comment by MikhailArkhipov** [comment]
Done? Close?
